### DISCUSSION TITLE
Option to set a manually compiled sumneko lua-language-server

### DISCUSSION THREAD

**Discussion by voyeg3r** [VonHeikemen/lsp-zero.nvim#25] [Ideas]
My machine is 32bit, so the `:LspInstall` command always fails, I manually compiled sumneko LUA and added it to the path, I just want to make the lspinstall aware of this. Would it be difficult?

The log is this:

```
Most likely the current operating system, architecture (x86), or libc (glibc) is not supported."
```

Probably the plugin already has this feature but I know very little of lua!
How would it be if I wanted only the sumneko lua manually installed?

```
-- my compiled version is here in my .zshenv:
-- [ -d ~/.config/nvim/lua-language-server ] && export PATH="$HOME/.config/nvim/lua-language-server/bin:$PATH"

lsp-zero.setup_servers()
```

Thanks for your great job!

**Comment by VonHeikemen** [comment]
I believe in this case you can ignore `LspInstall`. Configure the server with `lspconfig` directly.

The function `.build_options()` gives you the basic options for the server. And, `.defaults.nvim_workspace()` can add support for neovim's api.

Edit: you'll also need to add an extra source for `nvim-cmp`.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

-- Setup lua server for neovim
require('lspconfig').sumneko_lua.setup(
  lsp.build_options('sumneko_lua', lsp.defaults.nvim_workspace())
)

-- Add neovim api source to nvim-cmp
local cmp_sources = lsp.defaults.cmp_sources()
table.insert(cmp_sources, {name = 'nvim_lua'})
require('cmp').setup.filetype('lua', {sources = cmp_sources})

lsp.setup()
```
